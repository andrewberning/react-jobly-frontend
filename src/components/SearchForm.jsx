import { useState } from "react";
import "./SearchForm.css"

/** Search widget.
 *
 * Appears on CompanyList and JobList so that these can be filtered
 * down.
 *
 * This component doesn't *do* the searching, but it renders the search
 * form and calls the `searchFor` function prop that runs in a parent to do the
 * searching.
 *
 * { CompanyList, JobList } -> SearchForm
 */

export default function SearchForm({ searchFor }) {
  const [searchTerm, setSearchTerm] = useState("");

  /** Tell parent to filter */
  function handleSubmit(evt) {
    evt.preventDefault();
    searchFor(searchTerm.trim() || undefined);
    setSearchTerm(searchTerm.trim());
  }

  /** Update form fields */
  function handleChange(evt) {
    setSearchTerm(evt.target.value);
  }

  return (
    <div className="SerchForm mb-4">
      <form className="form-inline" onSubmit={handleSubmit}>
        <input
            className="form-control form-control-lg flex-grow-1"
            name="searchTerm"
            placeholder="Enter search term..."
            value={searchTerm}
            onChange={handleChange}
        />
        <button className="btn btn-lg btn-primary">Submit</button>
      </form>
    </div>
  );
}