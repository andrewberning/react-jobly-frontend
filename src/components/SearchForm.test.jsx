import { expect, it } from 'vitest';
import { render } from "@testing-library/react";
import '@testing-library/jest-dom';
import { MemoryRouter } from 'react-router-dom';
import SearchForm from "./SearchForm";

it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <SearchForm />
    </MemoryRouter>
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
      <MemoryRouter>
        <SearchForm />
      </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});