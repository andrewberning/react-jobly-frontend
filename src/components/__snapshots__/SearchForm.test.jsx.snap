// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="SerchForm mb-4"
  >
    <form
      class="form-inline"
    >
      <input
        class="form-control form-control-lg flex-grow-1"
        name="searchTerm"
        placeholder="Enter search term..."
        value=""
      />
      <button
        class="btn btn-lg btn-primary"
      >
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
